server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DLPConfigurationRoutesTest.java

```diff
@@ -0,0 +1,571 @@
+                "}";
+
+            given()
+                .body(body)
+            .when()
+                .put(DEFAULT_DOMAIN)
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .contentType(JSON_CONTENT_TYPE)
+                .body("statusCode", is(400))
+                .body("type", is("InvalidArgument"))
+                .body("message", is("JSON payload of the request is not valid"));
+        }
+
+        @Test
+        void putShouldReturnNotFoundWhenDomainNotInList() {
+            String body =
+                "[{" +
+                "  \"id\": \"1\"," +
+                "  \"expression\": \"expression 1\"," +
+                "  \"explanation\": \"explanation 1\"," +
+                "  \"targetsSender\": true," +
+                "  \"targetsRecipients\": true," +
+                "  \"targetsContent\": true" +
+                "}," +
+                "{" +
+                "  \"id\": \"2\"," +
+                "  \"expression\": \"expression 2\"," +
+                "  \"explanation\": \"explanation 2\"," +
+                "  \"targetsSender\": false," +
+                "  \"targetsRecipients\": false," +
+                "  \"targetsContent\": false" +
+                "}]";
+
+            given()
+                .body(body)
+            .when()
+                .put("strange.com")
+            .then()
+                .statusCode(HttpStatus.NOT_FOUND_404)
+                .contentType(JSON_CONTENT_TYPE)
+                .body("statusCode", is(HttpStatus.NOT_FOUND_404))
+                .body("type", is("InvalidArgument"))
+                .body("message", is("'strange.com' is not managed by this James server"));
+        }
+
+        @Test
+        void putShouldReturnBadRequestWhenDomainIsNotValid() {
+            String body =
+                "[{" +
+                "  \"id\": \"1\"," +
+                "  \"expression\": \"expression 1\"," +
+                "  \"explanation\": \"explanation 1\"," +
+                "  \"targetsSender\": true," +
+                "  \"targetsRecipients\": true," +
+                "  \"targetsContent\": true" +
+                "}," +
+                "{" +
+                "  \"id\": \"2\"," +
+                "  \"expression\": \"expression 2\"," +
+                "  \"explanation\": \"explanation 2\"," +
+                "  \"targetsSender\": false," +
+                "  \"targetsRecipients\": false," +
+                "  \"targetsContent\": false" +
+                "}]";
+
+            given()
+                .body(body)
+            .when()
+                .put("dr@strange.com")
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .contentType(JSON_CONTENT_TYPE)
+                .body("statusCode", is(HttpStatus.BAD_REQUEST_400))
+                .body("type", is("InvalidArgument"))
+                .body("message", is("Invalid request for domain: dr@strange.com"));
```

Comment:
Again I'd like the cause to explain me why the request is not valid...

Reply:
The exception we catch here is `IllegalArgumentException` throw by `Preconditions` check, and the exception doesn't have a message then`cause` field is null, i can add one statement to assert cause is null

Reply:
Then it is not OK!

We need to tell why we are failing: Ints: *the domain is not valid*. Why don't we pass the cause to the `ErrorResponder` ?